File: apps/api/src/config/env.ts
import 'dotenv/config'
import { z } from 'zod'

const Env = z.object({
  NODE_ENV: z.enum(['development','test','production']).default('development'),
  PORT: z.coerce.number().default(8080),
  DATABASE_URL: z.string(),
  CORS_ORIGIN: z.string().default('*'),
  ARCHIVE_AFTER_MIN: z.coerce.number().default(30)
})

export const env = Env.parse(process.env)

File: apps/api_/src/config/env.py
import os
from pydantic_settings import BaseSettings
from enum import Enum
from typing import Optional

class NodeEnv(str, Enum):
    DEVELOPMENT = "development"
    TEST = "test"
    PRODUCTION = "production"

class Settings(BaseSettings):
    NODE_ENV: NodeEnv = NodeEnv.DEVELOPMENT
    PORT: int = 8080
    # Make DATABASE_URL optional to allow app to start without DB configured
    DATABASE_URL: Optional[str] = None
    CORS_ORIGIN: str = "*"
    ARCHIVE_AFTER_MIN: int = 30
    # Accept lowercase values from .env; logger will normalize
    LOG_LEVEL: str = "INFO"

    class Config:
        env_file = ".env"
        case_sensitive = False

env = Settings()
